tests/dbshell/test_postgresql.py

```diff
@@ -24,6 +24,27 @@ def _mock_subprocess_run(*args, env=os.environ, **kwargs):
             DatabaseClient.runshell_db(dbinfo)
         return self.subprocess_args, self.pgpassword
 
+    def _run_it_all_env(self, dbinfo):
+        """
+        That function invokes the runshell command, while mocking
+        subprocess.run(). It returns a 2-tuple with:
+        - The command line list
+        - The list of values of the PGP% environment variables, or [].
+        """
+        def _mock_subprocess_run(*args, env=os.environ, **_kwargs):
+            self.subprocess_args = list(*args)
+            # Get all postgres related env
+            # Full list can be found at https://www.postgresql.org/docs/current/libpq-envars.html
+            self.pg_env_keys = [e for e in env if e.startswith('PG')]
+            # Sort environment vars to preserve its order
+            self.pg_env_keys.sort()
+            # Get values in form ENV_NAME=ENV_VALUE
+            self.pg_env = [itm + '=' + env[itm] for itm in self.pg_env_keys]
```

Comment:
Returning a dict would save you the sorting and the string parsing later in the tests, and environment variables are not sensitive to ordering, anyways.

Reply:
Thank you for your advise! I've made correction in my code.